### ISSUE TITLE
[bug] `@format` in list not working (DjangoDynaconf)

### ISSUE THREAD

**Issue by wengole** [rochacbruno/dynaconf#402]
**Describe the bug**
The `@format` tag doesn't appear to be working in a list of strings (in TOML). Possibly related to use of `dynaconf_merge` at the same time. Also possibly related to `DjangoDynaconf` use

**To Reproduce**

1. Having the following config files:

<details>
<summary> Config files </summary>

**settings.toml**
```toml
[default]
site_domain = ""
allowed_hosts = ["localhost", "127.0.0.1"]

[development]
allowed_hosts = ["@format {this.site_domain}", "dynaconf_merge"]
```

</details>

2. Having the following app code:

<details>
<summary> Code </summary>

**settings.py**
```python
...
# HERE STARTS DYNACONF EXTENSION LOAD (Keep at the very bottom of settings.py)
# Read more at https://dynaconf.readthedocs.io/en/latest/guides/django.html
import dynaconf  # noqa

settings = dynaconf.DjangoDynaconf(__name__)  # noqa
```

</details>

3. In shell or at runtime (viewing settings with djdt) shows the following

<details>
<summary> Execution </summary>

```python
ALLOWED_HOSTS = ['localhost', '127.0.0.1', '@format {this.site_domain}']
```

</details>

**Expected behavior**
The value `"@format {this.site_domain}"` should be interpolated as an empty string (as per the example settings.toml above)

**Environment (please complete the following information):**
 - OS: Linux
 - Dynaconf Version 3.1.0
 - Frameworks in use: Django 3.0.7

**Comment by rochacbruno** [comment]
duplicate of #392 

I worked a bit on it, found where the problem is but had no time yet to fix so probably this fix will not get in for this next release.

**Comment by rochacbruno** [comment]
fixed on #405
